### PR TITLE
#158354354 Launch add asset form

### PR DESCRIPTION
**Pull request by Billmike**
#### What does this PR do?
Creates the functionality for launching the add asset form.

#### Description of tasks to be completed?
- write method handler for changing the class of the continue button
- write method handler for listening for change events on the select tag
- write method handler for displaying asset form
- modify addassetpane.scss file to reflect new changes

#### How should this be manually tested?
- Run yarn start:dev to start the dev server.
- Sign in with your Andela account. Click on the caret close to the displayed username.
- Click on the `Add Asset` button to open the Add asset pane.
- The `Continue` button should be disabled.
- Select an option, and the continue button should now be active.

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
[#158354354](https://www.pivotaltracker.com/story/show/158354354)

#### Screenshots
screen shot 2018-06-26 at 9 21 16 am

screen shot 2018-06-26 at 9 21 22 am

### CODE REVIEW

File: src/components/AddAssetPane/index.tsx [hunk 1]
```diff
@@ -10,31 +10,90 @@ import Button from 'components/Button';
...
+
+  /**
+   * This method handler is responsible for changing the appearance of the Continue button.
+   * Depending on the value of the state.value key, it changes the css class passed to it.
+   * 
+   * @param {string}
+   * 
+   * @returns {string}
+   */
+  changeButtonClass = (currentValue) => {
```
Comment (Dammyy) [inline src/components/AddAssetPane/index.tsx]: Please add the **argumentName**  and **returnVariable** . Also add a description for them.  You can refer to the wiki for more information about the JSDoc format we use in documenting functions. https://github.com/andela/activo-web/wiki/Best-practices-for-writing-a-function

Comment (Billmike) [reply src/components/AddAssetPane/index.tsx]: Fixed.

**Comment by kevinejiro** [comment]
Please resolve conflicts

**Comment by Billmike** [comment]
@kevinejiro Fixed.

File: src/components/AddAssetPane/AddAssetPane.test.tsx [hunk 2]
```diff
@@ -47,4 +47,15 @@ describe('AddAssetPane component', () => {
...
+    const wrapper = shallow(<AddAssetPane { ...props } />);
+
+    wrapper.find('select').simulate('change', {
+      target: {
+        value: 'Chairs',
+      },
+    });
+    expect(wrapper.state().value).toEqual('Chairs');
+  });
 });
```
Comment (missating) [inline src/components/AddAssetPane/AddAssetPane.test.tsx]: @Billmike I think it would also be awesome if you add a test for when the `continue` button becomes active after an asset has been selected.

File: src/components/AddAssetPane/index.tsx [hunk 3]
```diff
@@ -10,33 +10,89 @@ import Button from 'components/Button';
...
+    this.setState({ showAssetForm: !this.state.showAssetForm });
+  }
+
+  render() {
+    return (
+      <div className={this.props.showAssetPane ? 'asset-pane asset-pane--active' : 'asset-pane'}>
       <h2 className="asset-pane__heading"> ADD NEW ASSET </h2>
-      <span className="asset-pane__close-icon" onClick={props.onClick}>
+      <span className="asset-pane__close-icon" onClick={this.props.onClick}>
         &times;
```
Comment (missating) [inline src/components/AddAssetPane/index.tsx]: Why did you add `this`?

File: src/components/AddAssetPane/index.tsx [hunk 4]
```diff
@@ -10,33 +10,89 @@ import Button from 'components/Button';
...
-        <Button name="CONTINUE" size="regular blue" />
-        <Button id="cancel-asset-button" name="CANCEL" size="regular white" onClick={props.onClick}/>
+        <Button
+          name="CONTINUE"
+          size={`${this.changeButtonClass(this.state.value)} blue`}
+          onClick={this.toggleAssetFormVisibility}
+        />
+        <Button id="cancel-asset-button" name="CANCEL" size="regular white" onClick={this.props.onClick} />
       </div>
     </div>
```
Comment (missating) [inline src/components/AddAssetPane/index.tsx]: These `Buttons` are not present on the pane where the form is to be rendered. The form design comes with an entirely new set of buttons, but with this implementation, the buttons for pane are on the form too.

Comment (Billmike) [reply src/components/AddAssetPane/index.tsx]: @missating So I left the buttons the way it was so as not to disrupt the design we currently have now. The design for the Add asset form is a separate task, and would require its own buttons like you've rightly stated. Also, I require the button to be the way it is to confirm that the state is toggled whenever the continue button is clicked.

File: src/components/AddAssetPane/AddAssetPane.test.tsx (hunk 2 above)
Comment (Billmike) [reply src/components/AddAssetPane/AddAssetPane.test.tsx]: @missating That would be repeating this test again, since the button only becomes active when the state changes with a valid option, like is already being tested for above. Do you have any suggestion in mind as to what I could test differently while checking that the continue button is active after an asset is selected?

Comment (missating) [reply src/components/AddAssetPane/AddAssetPane.test.tsx]: Yes. the button becomes active only when the state changes but what I'm saying is that there is no test to show that the button is now active after the state changes. You get?

or perhaps is it possible to add that to this test i.e the button becoming active?

Comment (Billmike) [reply src/components/AddAssetPane/AddAssetPane.test.tsx]: Perhaps what you mean is writing an assertion to confirm that the button becomes active when the state changes. 'Cause writing a separate test for that would just be repetition and that's not DRY.
